File: solutions/elixir/rna-transcription/1/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RnaTranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    rna =  %{ 65 => 85, 84 => 65, 71 => 67, 67 => 71}
    dna
    |> Enum.map(fn ch -> rna[ch] end)
  end
end

File: solutions/julia/rna-transcription/1/rna-transcription.jl
"""
    to_rna(dna)

The transcribed RNA strand given the DNA strand dna

"""
function to_rna(dna)
    rna = ""
    for n in dna
        if n == 'G'
            rna = rna*'C'
        elseif n == 'C'
            rna = rna*'G'
        elseif n == 'T'
            rna = rna*'A'
        elseif n == 'A'
            rna = rna*'U'
        else
            throw(ErrorException("Invalid input."))
        end
    end
    rna
end

